pkg/build/buildutil/util_test.go

```diff
@@ -8,6 +8,37 @@ import (
 	corev1 "k8s.io/api/core/v1"
 )
 
+func TestLabelValue(t *testing.T) {
+	testCases := []struct {
+		name           string
+		input          string
+		expectedOutput string
+	}{
+		{
+			name:           "do-not-end-with-hyphen",
+			input:          "cluster-kube-controller-manager-operator-4.3.0.ipv6-2019-11-27-0001-build",
+			expectedOutput: "cluster-kube-controller-manager-operator-4.3.0.ipv6-2019-11-27",
+		},
+		{
+			name:           "do-not-start-with-hyphen",
```

Comment:
this test case would go away if the "it is based on the build config name which has already been validated, etc." notion is agreed upon.